web/firefoxcom.js

```diff
@@ -285,6 +285,47 @@ class MozL10n {
   window.addEventListener("editingaction", handleEvent);
 })();
 
+if (
+  typeof PDFJSDev === "undefined"
+    ? window.isGECKOVIEW
+    : PDFJSDev.test("GECKOVIEW")
+) {
+  (function listenQueryEvents() {
+    window.addEventListener("pdf.js.query", async evt => {
+      const {
+        detail: { queryId },
+      } = evt;
+      let result = null;
+      if (queryId === "canDownloadInsteadOfPrint") {
+        result = false;
+        const { pdfDocument, pdfViewer } = PDFViewerApplication;
+        if (pdfDocument) {
+          try {
+            const hasModifiedAnnotations =
+              pdfDocument.annotationStorage.size > 0;
+            // WillPrint is called just before printing the document and could
+            // lead to have modified annotations.
+            const hasWillPrint =
+              pdfViewer.enableScripting &&
+              !!(await pdfDocument.getJSActions())?.WillPrint;
```

Comment:
Previously we'd only call this method *once* in the viewer, so should we perhaps start caching this in the API to avoid having to do a round-trip to the worker-thread during printing?

Doing should be quite simple, since you'd only need to replace
https://github.com/mozilla/pdf.js/blob/5b8f6804809a36ce5c2426fa8772b5cb51f6620d/src/display/api.js#L3024
with
```javascript
return this.#cacheSimpleMethod("GetDocJSActions"); 
```